pandas/core/generic.py

```diff
@@ -6674,9 +6674,12 @@ def fillna(
                     if k not in result:
                         continue
                     downcast_k = downcast if not is_dict else downcast.get(k)
+                    # GH47649
                     result.loc[:, k] = (
                         result[k].fillna(v, limit=limit, downcast=downcast_k).values
                     )
+                    # TODO: Revert to result.loc[:, k] = result[k].fillna(...)
```

Comment:
Please put this right next to the todo and we want to use loc on both sides I think

Reply:
You mean make comments like this?
```
# TODO: result.loc[:, k] = result.loc[:, k].fillna(
#     v, limit=limit, downcast=downcast_k
# )
# Revert when GH45751 is fixed
```

Reply:
Thx